File: misc/psd-format/scripts/udp_receiver.pl
# udp_receiver.pl

use IO::Socket::INET;

# Turn on System variable for Buffering output
$| = 1;

$port= 5000;

# Create a a new socket
$socket=new IO::Socket::INET->new(LocalPort=>$port,Proto=>'udp');

print "\nUDP Server: waiting for client on port $port ...";
$cnt= 1;

while(1)
{
	$socket->recv($buffer,1024);
	$n= length $buffer;
	$peer_address = $socket->peerhost();
	$peer_port = $socket->peerport();

	print "\n $cnt ($peer_address:$peer_port) received : $n bytes\n";

	$buffer =~ s/(.)/sprintf("%02X ",ord($1))/eg;
	print $buffer;

	$cnt++;
}

File: misc/psd-format/scripts/udp_receiver.py
# udp_receiver.py

import socket
port = 5000
server_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
server_socket.bind(("", port))

print"UDP server: waiting for client on port ", port, "..."
cnt = 1

while 1:
	data, address = server_socket.recvfrom(256)
	print cnt, "( " ,address[0], " " , address[1] , " ) "
	cnt = cnt + 1

